src/netlink.c

```diff
@@ -503,18 +521,26 @@ static int qos_filter_cb(const struct nlmsghdr *nlh, void *args) {
       return MNL_CB_ERROR;
     }
 
-    mnl_attr_parse_nested(attr, qos_attr_cb, &bs);
+    mnl_attr_parse_nested(attr, qos_attr_cb, &stats);
 
-    if (bs != NULL) {
+    if (stats.bstats != NULL || stats.qstats != NULL) {
       char type_instance[DATA_MAX_NAME_LEN];
 
       stats_submitted = 1;
 
       ssnprintf(type_instance, sizeof(type_instance), "%s-%s", tc_type,
                 tc_inst);
-
-      submit_one(dev, "ipt_bytes", type_instance, bs->bytes);
-      submit_one(dev, "ipt_packets", type_instance, bs->packets);
+      if (stats.bstats != NULL) {
+        submit_one(dev, "ipt_bytes", type_instance, stats.bstats->bytes);
+        submit_one(dev, "ipt_packets", type_instance, stats.bstats->packets);
+      }
+      if (stats.qstats != NULL) {
+        submit_one(dev, "ipt_qlen", type_instance, stats.qstats->qlen);
+        submit_one(dev, "ipt_backlog", type_instance, stats.qstats->backlog);
```

Comment:
Do I understand this correctly that "qlen" is the length of the queue and "backlog" the number of packets in the queue? If so, I think it'd make most sense to do *avail = qlen - backlog* and then submit *avail* and *backlog* as *queue_length-free* and *queue_length-used*.